lambdas/service/app.py

```diff
@@ -367,7 +367,7 @@ def link(self, *, previous: bool, **params: str) -> Optional[str]:
                     'order': self.order,
                     'size': self.size
                 }
-            return self.self_url + '?' + urllib.parse.urlencode(params)
+            return self.self_url.set(query_params=params)
```

Comment:
This modifies the `furl` instance referenced by `self.self_url`. We usually don't want a getter method to have such a side effect. Please also review all other invocations of `furl.set` in this respect. Some are safe, some aren't and require a copy to me made beforehand.